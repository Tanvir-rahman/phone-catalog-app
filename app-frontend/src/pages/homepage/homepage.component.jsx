import React, { useEffect, lazy, Suspense } from 'react';
import { connect  } from 'react-redux';

import './homepage.styles.scss';

import { fetchPhoneList } from '@store/phones/phone.actions';

import Spinner from '@components/spinner/spinner.component';

const PhoneListContainer = lazy(() =>
  import('@components/phone-list/phone-list.container')
);

export const HomePage = ({ fetchPhoneList, match }) => {
  useEffect(() => {
    fetchPhoneList();
  }, [fetchPhoneList]);

  return (
    <Suspense fallback={<Spinner />}>
      <h1 className="title"> Phone Catalog App </h1>
      <PhoneListContainer />
    </Suspense>
  );
};

const mapDispatchToProps = dispatch => ({
  fetchPhoneList: () => dispatch(fetchPhoneList())
});

export default connect(
  null,
  mapDispatchToProps
)(HomePage);