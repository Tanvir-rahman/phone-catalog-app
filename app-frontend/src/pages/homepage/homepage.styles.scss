.title  {
  font-size: 32px;
  color: #333;
  text-align: center;
  margin: 20px 0 40px;
}