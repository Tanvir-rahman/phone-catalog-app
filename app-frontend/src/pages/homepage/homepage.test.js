import React from 'react';
import { shallow } from 'enzyme';
import { HomePage } from './homepage.component';

describe('Homepage component', () => {
  let wrapper;
  let mockfetchPhoneList;

  beforeEach(() => {
    mockfetchPhoneList= jest.fn();

    const mockProps = {
      fetchPhoneList: mockfetchPhoneList
    };

    wrapper = shallow(<HomePage {...mockProps} />);
  });

  it('should render Homepage component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title in Title Container', () => {
    expect(wrapper.find('.title').text()).toBe(' Phone Catalog App ');
  });
});
