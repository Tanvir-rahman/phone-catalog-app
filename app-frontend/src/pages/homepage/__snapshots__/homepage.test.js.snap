// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component should render Homepage component 1`] = `
<Suspense
  fallback={<Spinner />}
  fetchPhoneList={[MockFunction]}
>
  <h1
    className="title"
  >
     Phone Catalog App 
  </h1>
  <lazy />
</Suspense>
`;
